aws/logs_monitoring/lambda_function.py

```diff
@@ -525,6 +530,30 @@ def extract_ddtags_from_message(event):
         event[DD_CUSTOM_TAGS] = f"{event[DD_CUSTOM_TAGS]},{extracted_ddtags}"
 
 
+def extract_arn_hostname(message):
+    """Extract the hostname from userIdentity.arn field if it matches AWS hostnames
+
+    >>> extract_arn_hostname(json.dumps(
+    >>> {"usertIdentity": {"arn": "arn:aws:sts::123456789123:assumed-role/DatadogAWSIntegrationRole/i-08014e4f62ccf762d"}})
+    >>> )
+    i-0123456789abcdef0
+    >>> extract_arn_hostname({"message": "test"})
+    None
+    """
+    if message is not None:
+        if isinstance(message, str):
+            try:
+                message = json.loads(message)
+            except json.JSONDecodeError:
```

Comment:
I would also add some debugging logs in this function to help troubleshooting when needed, e.g., https://github.com/DataDog/datadog-serverless-functions/blob/3d061da905ffaa50acecce06f7f6f0c5dde3acf7/aws/logs_monitoring/lambda_function.py#L405